internal/config/config.go

```diff
@@ -0,0 +1,58 @@
+package config
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/structs"
+	"github.com/knadh/koanf/v2"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/client"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/logger"
+	"github.com/tidwall/pretty"
+)
+
+type (
+	// Config holds all configurations.
+
+	Config struct {
+		Logger logger.Config `json:"logger,omitempty" koanf:"logger"`
+		NATS   client.Config `json:"nats,omitempty" koanf:"nats"`
+	}
+)
+
+// New reads configuration with koanf.
+func New() Config {
+	var instance Config
+
+	k := koanf.New(".")
+
+	// load default configuration from file
+	if err := k.Load(structs.Provider(Default(), "koanf"), nil); err != nil {
+		log.Fatalf("error loading default: %s", err)
+	}
+
+	if err := k.Load(file.Provider("config.yaml"), yaml.Parser()); err != nil {
+		log.Printf("error loading config.yaml")
+	}
+
```

Comment:
We can use environment variables as another source for our configuration but it is not required because we already have file.

Reply:
Fixed. 👍